src/Type.cc

```diff
@@ -1116,6 +1126,59 @@ class VectorFieldInit final : public FieldInit
 
 	} // namespace detail
 
+// Helper TraversalCallback optimizing RecordType instances by moving
+// deferrable FieldInits from creation_inits to deferred_inits once
+// parsing has completed.
+class RecordType::CreationInitsOptimizer : public detail::TraversalCallback
+	{
+public:
+	detail::TraversalCode PreID(const detail::ID* id) override
+		{
+		if ( const auto& t = id->GetType() )
+			HANDLE_TC_TYPE_POST(t->Traverse(this));
+
+		return detail::TC_CONTINUE;
+		}
```

Comment:
`PreType()` is somewhat recent. `traverse_all()` will not by default invoke `PreType()`, so that `PreID()` trampoline is needed somewhere. It took quite a while to figure out and am now unsure if it's not bordering unexpected behavior.

In any case, I've switched to `PreTypedef()` after looking again. That's invoked for all IDs that are types, ends-up shorter and coincidentally/accidentally also allows to get rid of the `const_cast`.